MoreLinq/Assignment.cs

```diff
@@ -0,0 +1,29 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    static class Assignment
+    {
+        public static T Set<T>(ref T slot, T newValue)
```

Comment:
This (undocumented) method serves to elide one line in each call location. Is it necessary?

Reply:
It allows to compress three lines into one in a reliable way. Nothing is ever necessary but it helped me to encode a pattern in a reusable manner because it felt dirty to have the same thing twice, once in `OnError` and another time in `OnCompleted`. That said, I found that there was a lot more duplication between the two and so I decided to consolidate them entirely and now `Assignment.Set` is gone with 850b16f1aadc35b90f716909abe0225dc80934fe.